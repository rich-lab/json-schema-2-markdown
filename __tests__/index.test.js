const { jsonschema2md } = require('../lib')

it('simple', async () => {
  const src = {
    schemaPath: 'fixtures/simple.json',
    outputPath: 'docs/config/README.md',
  }

  const ret = await jsonschema2md({
    src,
    write: false,
    cwd: __dirname
  })

  expect(ret).toMatchSnapshot()
})

it('doc & i18n', async () => {
  const src = {
    locale: 'zh-CN',
    schemaPath: 'fixtures/doc.json',
    outputPath: 'docs/config/README.md',
  }

  const ret = await jsonschema2md({
    src,
    write: false,
    cwd: __dirname
  })

  expect(ret).toMatchSnapshot()
})

it('merge md', async () => {
  const src = {
    locale: 'zh-CN',
    schemaPath: 'fixtures/merge-md.json',
    outputPath: 'docs/config/README.md',
  }

  const ret = await jsonschema2md({
    src,
    write: false,
    cwd: __dirname
  })

  expect(ret).toMatchSnapshot()
})
