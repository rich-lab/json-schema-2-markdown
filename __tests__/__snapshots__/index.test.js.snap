// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doc & i18n 1`] = `
"# foo.config.js
 
My Config File

## type

- 类型: \`\\"foo\\" | \\"bar\\"\`
- 描述: TYPE
- 昵称: ulivz

## env

- 类型: \`string\`
- 描述: ENV"
`;

exports[`merge md 1`] = `
"# foo.config.js
 
My Config File

## type

- 类型: \`\\"foo\\" | \\"bar\\"\`
- 描述: TYPE's description


XXXXXXXX type XXXXXXXX

## env

- 类型: \`string\`
- 描述: ENV's description


XXXXXXXX env XXXXXXXX"
`;

exports[`simple 1`] = `
"# foo.config.js
 
My Config File

## type

- Type: \`\\"foo\\" | \\"bar\\"\`
- Description: TYPE's description

## env

- Type: \`string\`
- Description: ENV's description"
`;
